public/src/client/topic/posts.js

```diff
@@ -33,10 +33,12 @@ define('forum/topic/posts', [
 		ajaxify.data.postcount ++;
 		postTools.updatePostCount(ajaxify.data.postcount);
 
-		if (config.usePagination) {
-			onNewPostPagination(data);
-		} else {
-			onNewPostInfiniteScroll(data);
+		if (config.scrollToMyPost) {
```

Comment:
Hmm I don't see anywhere that you pass scrollToMyPost the client. We pass them to the client here https://github.com/NodeBB/NodeBB/blob/master/src/controllers/api.js#L73 if we need the setting to be available in `config.*` but in this case we only need it in the topic view, that's why I suggested adding it to the topics controller.

I can make the change after I merge.